File: programmers/js/level00/120894.js
// 영어가 싫어요
function solution(numbers) {
  const numArr = ["zero", "one", "two", "three", "four", 
                  "five", "six", "seven", "eight", "nine"];
  return +numArr
    .reduce((acc, cur) => acc.split(cur).join(numArr.indexOf(cur)), numbers);
}

// 다른 풀이
// function solution(numbers) {
//   let numStr = ["zero", "one", "two", "three", "four", 
//                 "five", "six", "seven", "eight", "nine" ];

//   numStr.forEach((str, idx) => {
//     numbers = numbers.replaceAll(str, idx);
//   });

//   return Number(numbers);
// }

File: programmers/java/level00/120894.java
// 영어가 싫어요

class Solution {
    public long solution(String numbers) {
        String[] numArray = {"zero", "one", "two", "three", "four", 
                       "five", "six", "seven", "eight", "nine"};
        
        for (int i = 0; i < numArray.length; i++) {
            numbers = numbers.replace(numArray[i], String.valueOf(i));
        }
        
        return Long.valueOf(numbers);
    }
}